### PR TITLE
(WIP) Paypal Payout Report Integration

### PR DESCRIPTION
**Pull request by yachtcaptain23**
## Pull Request Name

[Add screenshot if applicable]

#### Features

Enter a brief summary of the Pull Request - Features, motivations, and issues it #closes

#### How To Test

1. Step one
2. Step two
3. Step three

#### Pull Request Checklist

- [ ] Adequate test coverage exists to prevent regressions -- [Guide to Testing](https://guides.rubyonrails.org/testing.html)
- [ ] XSS is mitigated -- [Guide to XSS Prevention](https://guides.rubyonrails.org/security.html#cross-site-scripting-xss)
- [ ] No raw SQL -- [Guide to SQL Injection Prevention](https://guides.rubyonrails.org/security.html#sql-injection)
- [ ] UI/UX is responsive -- [Guide to Responsive UI/UX](https://developers.google.com/web/fundamentals/design-and-ux/responsive/)
- [ ] Integrated Matomo for new UI elements -- [Guide to Matomo](https://developer.matomo.org/guides/integrate-introduction)
- [ ] Passes checklist for Progressive Web App -- [Guide to PWAs](https://developers.google.com/web/progressive-web-apps/checklist)

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -276,6 +276,23 @@
...
+    t.text "country"
+    t.boolean "verified_account"
+    t.text "paypal_account_id"
+    t.boolean "hidden", default: false
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["user_id"], name: "index_paypal_connections_on_user_id"
+  end
+
+  end
```
Comment (corymcdonald) [inline db/schema.rb]: Seems like this isn't the most up to date schema? Can you make sure this aligns when you drop/recreate your DB like `rails db:drop db:create db:migrate`?

File: app/views/publishers/home.html.slim [hunk 2]
```diff
@@ -98,62 +98,11 @@ script type="text/html" id="confirm_default_currency_modal_wrapper"
...
-                    span.bat-amount#last_deposit_bat_amount= last_deposit_balance
-                    - unless last_deposit_balance == I18n.t("helpers.publisher.no_deposit")
-                      span.bat-code= " BAT"
-                  .converted#last_deposit_converted_amount
-                    = publisher_converted_last_settlement_balance(@publisher)
-                = link_to(t(".uphold.check_balance"), uphold_dashboard_url, class: "btn btn-primary",
-                        id: "uphold_check_balance", :"data-piwik-action" => "CheckUpholdBalanceClicked",
-                        :"data-piwik-name" => "Clicked", :"data-piwik-value" => "Dashboard")
+      .col-md.mb-4
+        - if !(@publisher.excluded_from_payout? || @publisher.paypal_locale?(params[:locale]))
```
[the remark this answers is not in the document]
Comment (yachtcaptain23) [reply app/views/publishers/home.html.slim]: Covered on line `104`

File: config/routes.rb [hunk 3]
```diff
@@ -29,6 +29,12 @@
         resource :two_factor_authentications_removal
       end
 
+      resources :paypal_accounts, controller: "publishers/paypal_accounts", only: [] do
+        get :connect_callback, on: :collection
+        get :refresh
+        patch :disconnect
```
[the remark this answers is not in the document]
Comment (yachtcaptain23) [reply config/routes.rb]: Resolved offline

File: app/controllers/publishers/paypal_accounts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+require "faraday"
+
+module Publishers
+  class PaypalAccountsController < ApplicationController
+    before_action :authenticate_publisher!
+
+    def disconnect
+      current_publisher.paypal_connections.active.update_all(hidden: true)
```
[the remark this answers is not in the document]
Comment (yachtcaptain23) [reply app/controllers/publishers/paypal_accounts_controller.rb]: I fixed in the model where `has_one` is only referring to 1 active paypal connection

**Comment by yachtcaptain23** [comment]
Let's add to Paypal account information to PotentialPayment

File: db/schema.rb [hunk 5]
```diff
@@ -274,8 +274,25 @@
...
```
Comment (yachtcaptain23) [inline db/schema.rb]:
```suggestion
```

**Comment by yachtcaptain23** [comment]
Merged in https://github.com/brave-intl/publishers/pull/2494

**Pull request closed without merging**